qiskit/providers/ibmq/managed/utils.py

```diff
@@ -110,7 +127,19 @@ def format_job_details(
 
         job = mjob.job
         report.append("    job ID: {}".format(job.job_id()))
-        report.append("    name: {}".format(job.name()))
+
+        job_name_info = "    name: {}".format(job.name())
+        # If a job name does not match the managed job set name, mark it with an asterisk.
+        matched = JOB_SET_NAME_RE.match(job.name()) if job.name() else None
+        if not matched:
+            raise IBMQJobManagerInvalidStateError(
+                'Job {} is tagged for the job set {} but does not appear '
+                'to belong to the set.'.format(job.job_id(), managed_jobs_id))
```

Comment:
Unmatched job names should be handled elsewhere as this function is only used to format a report.